technical-docs/K8SSANDRA-OPERATOR-DESIGN.md

```diff
@@ -0,0 +1,410 @@
+# K8ssandra Operator Design
+
+Author: @jsanda
+
+## Overview
+
+GH issue: https://github.com/k8ssandra/k8ssandra/issues/683
+
+This is a design document for an operator for k8ssandra. For background see 
+https://github.com/k8ssandra/k8ssandra/issues/485.
+
+Much of the behavior that is implemented in the Helm charts will be reimplemented in various controllers. The role of the Helm charts changes to focus primarily on installation and configuration of operators.
+
+While we will continue to use and support Helm, it will not be required for installing k8ssandra from a purely technical perspective. Users can install manifests with kubectl apply for example. We can also provide support for kustomize.
```

Comment:
Given that kubectl ships with kustomize these days (kubectl apply -k), it's kinda given we should design our CRDs in a way that allows to use kustomize generators / transformers.